spec/lib/hysync/marc_synchronizer/runner_spec.rb

```diff
@@ -12,4 +12,176 @@
+      let(:hyacinth_record) do
+        record = described_class.default_digital_object_data
+        record['dynamic_field_data']['marc_005_last_modified'] = [{'marc_005_last_modified_value' => different_005}]
+        record
+      end
+      it { is_expected.to be true }
+    end
+    context 'hyacinth and marc 005 do not differ' do
+      it { is_expected.to be false }
+    end
+  end
+  describe '#reconcile_identifiers!' do
+    let(:runner) { described_class.allocate }
+    let(:default_data) { described_class.default_digital_object_data }
+    let(:marc_record) do
+      FactoryBot.build(:marc_record)
+    end
+    let(:marc_ids) { [] }
+    let(:marc_hyacinth_record) do
+      Hysync::MarcSynchronizer::MarcHyacinthRecord.new(marc_record, [], default_data.merge('identifiers' => marc_ids.dup))
+    end
+    let(:hyacinth_ids) { [] }
+    let(:existing_hyacinth_record) do
+      record = described_class.default_digital_object_data.merge('identifiers' => hyacinth_ids.dup)
+      record
+    end
+    before do
+      runner.reconcile_identifiers!(marc_hyacinth_record, existing_hyacinth_record)
+    end
+    context 'marc is superset of hyacinth' do
+      let(:marc_ids) { ['abc123', 'abc456'] }
+      let(:hyacinth_ids) { ['abc456'] }
+      let(:expected) { marc_ids }
+      it 'preserves marc identifiers in order' do
+        expect(marc_hyacinth_record.digital_object_data['identifiers']).to eql(expected)
+      end
+    end
+    context 'hyacinth is superset of marc' do
+      let(:marc_ids) { ['abc456'] }
+      let(:hyacinth_ids) { ['abc123', 'abc456'] }
+      let(:expected) { ['abc456', 'abc123'] }
+      it 'reorders the union to match marc order first' do
+        expect(marc_hyacinth_record.digital_object_data['identifiers']).to eql(expected)
+      end
+    end
+    context 'marc and hyacinth identifier sets are equal' do
+      let(:marc_ids) { ['abc123', 'abc456'] }
+      let(:hyacinth_ids) { marc_ids.reverse }
+      let(:expected) { marc_ids }
+      it 'reorders the union to match marc order' do
+        expect(marc_hyacinth_record.digital_object_data['identifiers']).to eql(expected)
+      end
+    end
+    context 'marc and hyacinth identifier sets are disjoint' do
+      let(:marc_ids) { ['abc123', 'abc456'] }
+      let(:hyacinth_ids) { ['abc789', '123abc'] }
+      let(:expected) { marc_ids + hyacinth_ids }
+      it 'assigns the union of values, marc first' do
+        expect(marc_hyacinth_record.digital_object_data['identifiers']).to eql(expected)
+      end
+    end
+  end
+  describe '#reconcile_projects!' do
+    let(:runner) { described_class.allocate }
+    let(:hyacinth_project) { 'hyacinth_only' }
+    let(:default_data) { described_class.default_digital_object_data }
+    let(:marc_record) do
+      record = FactoryBot.build(:marc_record)
+      marc_ids.each do |key|
+        record.append(MARC::DataField.new('965', ' ',  ' ', ['a', "965#{key}"]))
+      end
+      record
+    end
+    let(:marc_ids) { [] }
```

Comment:
`marc_ids` is a little unclear for the same reason as hyacinth_ids, below.  Seems like it's referring to object identifiers rather than project keys like in the `reconcile_identifiers!` method tests above.

Reply:
I'm pushing a commit w/ variable name changes and comments for these two variables

Reply:
Thanks!